SQF-VM/SQF.h

```diff
@@ -39,4 +40,20 @@ void destroy_command(PCMD command);
 PCMD find_command(PVM vm, const char* name, char type);
 PCMD find_type(PVM vm, const char* name);
 
+VALUE addScalarPointer(const PVALUE left, const PVALUE right);
+VALUE addScalar(const VALUE left, const VALUE right);
+VALUE multiplyScalarPointer(const PVALUE left, const PVALUE right);
+VALUE multiplyScalar(const VALUE left, const VALUE right);
+VALUE substractScalarPointer(const PVALUE left, const PVALUE right);
+VALUE substractScalar(const VALUE left, const VALUE right);
+VALUE divideScalarPointer(const PVALUE left, const PVALUE right);
+VALUE divideScalar(const VALUE left, const VALUE right);
+
+float dotProductPointer(const PARRAY leftArray, const PARRAY rightArray);
+float dotProduct(const ARRAY leftArray, const ARRAY rightArray);
+VALUE dotProductPointer_Value(const PARRAY leftArray, const PARRAY rightArray);
+VALUE dotProduct_Value(const ARRAY leftArray, const ARRAY rightArray);
+float vectorMagnitudePointer(const PARRAY array);
+float vectorMagnitude(const ARRAY array);
```

Comment:
theese should be moved into `SQF_types.h` (or alternative: create a new `SQF_scalar_type.h` specially for scalar and move ALL scalar related methods & structs to there)
also, scalar related methods should start with `scalar_`

Reply:
Also, please remove camel-case from theese methods

Reply:
`_Value` suffix should be lowercase